flink-runtime/src/main/java/org/apache/flink/runtime/state/RegisteredStateMetaInfoUtils.java

```diff
@@ -0,0 +1,176 @@
+                                .snapshotConfiguration();
+                break;
+            default:
+                throw new UnsupportedOperationException(
+                        "Unsupported state type to transform from org.apache.flink.runtime.state.RegisteredKeyValueStateBackendMetaInfo to org.apache.flink.runtime.state.v2.RegisteredKeyValueStateBackendMetaInfo");
+        }
+
+        if (!newStateType.equals(StateDescriptor.Type.MAP)) {
+            return new org.apache.flink.runtime.state.v2.RegisteredKeyValueStateBackendMetaInfo<>(
+                    stateMeta.getName(),
+                    newStateType,
+                    StateSerializerProvider.fromPreviousSerializerSnapshot(
+                            stateMeta.getTypeSerializerSnapshot(
+                                    StateMetaInfoSnapshot.CommonSerializerKeys
+                                            .NAMESPACE_SERIALIZER)),
+                    StateSerializerProvider.fromPreviousSerializerSnapshot(
+                            newStateSerializerSnapshot),
+                    StateSnapshotTransformer.StateSnapshotTransformFactory.noTransform());
+        } else {
+            return new org.apache.flink.runtime.state.v2
+                    .RegisteredKeyAndUserKeyValueStateBackendMetaInfo<>(
+                    stateMeta.getName(),
+                    newStateType,
+                    StateSerializerProvider.fromPreviousSerializerSnapshot(
+                            stateMeta.getTypeSerializerSnapshot(
+                                    StateMetaInfoSnapshot.CommonSerializerKeys
+                                            .NAMESPACE_SERIALIZER)),
+                    StateSerializerProvider.fromPreviousSerializerSnapshot(
+                            newStateSerializerSnapshot),
+                    StateSerializerProvider.fromPreviousSerializerSnapshot(
+                            newUserKeySerializerSnapshot),
+                    StateSnapshotTransformer.StateSnapshotTransformFactory.noTransform());
+        }
+    }
+
+    public static RegisteredKeyValueStateBackendMetaInfo createMetaInfoV1FromV2Snapshot(
+            StateMetaInfoSnapshot stateMeta) {
+
+        Preconditions.checkArgument(
+                stateMeta
+                        .getBackendStateType()
+                        .equals(StateMetaInfoSnapshot.BackendStateType.KEY_VALUE_V2));
+        org.apache.flink.api.common.state.StateDescriptor.Type newStateType;
+        TypeSerializerSnapshot oldValueSerializerSnapshot =
+                stateMeta.getTypeSerializerSnapshot(
+                        StateMetaInfoSnapshot.CommonSerializerKeys.VALUE_SERIALIZER);
+        TypeSerializerSnapshot oldUserKeySerializerSnapshot =
+                stateMeta.getTypeSerializerSnapshot(
+                        StateMetaInfoSnapshot.CommonSerializerKeys.USER_KEY_SERIALIZER);
+        ;
+        TypeSerializerSnapshot newUserKeySerializerSnapshot = oldValueSerializerSnapshot;
```

Comment:
How about renaming `newUserKeySerializerSnapshot` to `newStateSerializerSnapshot`?